docs: AWS codebuild: Avoid branch names like `main~3`.

janpapenbrock opened the pull request:
When the main branch has advanced compared to the currently checked out commit,`git name-rev` will print e.g. `main~3`.

To keep Cypress branch names clean, we want `COMMIT_INFO_BRANCH` to be just `main`, though.

Therefore, I suggest cutting off the suffix.

---

If this gets merged, I believe the docs have to be afterwards updated in multiple places, too, e.g. here: https://github.com/cypress-io/cypress-documentation/blob/main/docs/app/continuous-integration/aws-codebuild.mdx

I could do that, too, if desired, but of course don't have an overview of _all_ the correct places.

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/cypress-io/cypress-example-kitchensink?pullRequest=919) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/cypress-io/cypress-example-kitchensink?pullRequest=919) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/cypress-io/cypress-example-kitchensink?pullRequest=919) it.</sub>

MikeMcC399 commented:
@janpapenbrock 

Thanks for your PR!

Please change your commit message to start with `docs:` rather than `fix:` as the files you propose to change are not actually run in this repo, they are documentation only for AWS Codebuild. The same applies to your other PR https://github.com/cypress-io/cypress-example-kitchensink/pull/920.

If you have a public example using your amended AWS Codebuild workflow successfully running, that would be very helpful for review.

See https://github.com/cypress-io/cypress-example-kitchensink/blob/master/README.md#ci-workflow-examples

> This table shows additional examples of CI workflows. With the exception of GitHub Actions workflows, these are not live examples as in the table above and they may require modification before use.

I don't know why the CircleCI Pipeline is failing. That would need to be looked at by the Cypress.io team who have access to the CircleCI account.

MikeMcC399 commented:
@janpapenbrock 

You should probably also change the title of the PR

> fix: Avoid branches like main~3 for AWS codebuild

as well. ( `fix:` -> `docs:`)

janpapenbrock commented:
Changed commit message and PR title.

---

I even had some brackets wrong in the code, which I have now fixed.

---

I don't have a public workflow available.

The changes are somewhat testable in a shell, when "mocking" the output of `git rev-parse HEAD | xargs git name-rev` and leaving everything else the same:

```
#!/bin/bash

git_name_rev_output="some-hash remotes/origin/main~3"

# old code
export COMMIT_INFO_BRANCH="$(echo "$git_name_rev_output" | cut -d' ' -f2 | sed 's/remotes\/origin\///g')"
echo "old: $COMMIT_INFO_BRANCH"

# new code
export COMMIT_INFO_BRANCH="$(echo "$git_name_rev_output" | cut -d' ' -f2 | sed 's/remotes\/origin\///g' | sed 's/~.\+$//g')"
echo "new: $COMMIT_INFO_BRANCH"
```

Prints:
```
old: main~3
new: main
```
